discovery-infra/tests/base_test.py

```diff
@@ -0,0 +1,148 @@
+            vip_dhcp_allocation=EnvVariable.VIP_DHCP_ALLOCATION
+        )
+    
+    def delete_cluster_if_exists(self, api_client, cluster_name):
+        cluster_to_delete = self.get_cluster_by_name(
+            api_client=api_client, 
+            cluster_name=cluster_name
+        )
+
+        if cluster_to_delete:
+            api_client.delete_cluster(cluster_id=cluster_to_delete['id'])
+
+
+    def get_cluster_by_name(self, api_client, cluster_name):
+        clusters = api_client.clusters_list()
+
+        for cluster in clusters:
+            if cluster['name'] == cluster_name:
+                return cluster
+
+        return None
+
+    
+    def generate_and_download_image(self, cluster_id, api_client):
+        api_client.generate_and_download_image(
+            cluster_id=cluster_id, 
+            ssh_key=EnvVariable.SSH_PUBLIC_KEY, 
+            image_path=EnvVariable.ISO_DOWNLOAD_PATH
+        )
+
+    def wait_untill_hosts_are_discovered(self, cluster_id, api_client):
+        utils.wait_till_all_hosts_are_in_status(
+            client=api_client, 
+            cluster_id=cluster_id, 
+            nodes_count=EnvVariable.NUM_NODES, 
+            statuses=[consts.NodesStatus.PENDING_FOR_INPUT, consts.NodesStatus.KNOWN]
+        )
+    
+    def set_host_roles(self, cluster_id, api_client):
+        utils.set_hosts_roles_based_on_requested_name(
+            client=api_client,
+            cluster_id=cluster_id
+        )
+        
+    def set_ingress_and_api_vips(self, cluster_id, api_client, controler):
+        vips = controler.get_ingress_and_api_vips()
+        api_client.update_cluster(cluster_id, vips)
+
+    def start_cluster_install(self, cluster_id, api_client):
+        api_client.install_cluster(cluster_id=cluster_id)
+
+    def cancel_cluster_install(self, cluster_id, api_client):
+        api_client.cancel_cluster_install(cluster_id=cluster_id)
+
+    def wait_for_one_host_to_boot_durring_install(self, cluster_id, api_client):
+        utils.wait_till_at_least_one_host_is_in_status(
+            client=api_client,
+            cluster_id=cluster_id,
+            statuses=[consts.NodesStatus.INSTALLING_IN_PROGRESS]
+        )
+    
+    def assert_cluster_is_in_error_status(self, cluster_id, api_client):
+        assert utils.is_cluster_in_status(
+            client=api_client,
+            cluster_id=cluster_id, 
+            statuses=[consts.ClusterStatus.ERROR]
+        )
+    
+    def assert_cluster_is_in_cancelled_status(self, cluster_id, api_client):
+        assert utils.is_cluster_in_status(
+            client=api_client,
+            cluster_id=cluster_id, 
+            statuses=[consts.ClusterStatus.CANCELLED]
+        )
+    def reset_cluster_install(self, cluster_id, api_client):
+        api_client.reset_cluster_install(cluster_id=cluster_id)
+
+    def assert_cluster_is_in_insufficient_status(self, cluster_id, api_client):
+        assert utils.is_cluster_in_status(
```

Comment:
`assert` shouldn't be inside the helper functions. Just return the value and `assert` in the calling test

Reply:
done